backend/src/hmc-api/fetcher/utils.ts

```diff
@@ -0,0 +1,88 @@
+import * as APIv4 from "hyperschedule-shared/api/v4";
+import { DATA_DIR_PATH } from "hyperschedule-data";
+import type { Endpoint, Endpoints } from "./types";
+import { Params } from "./types";
+import { resolve, dirname } from "path";
+import { readFile, writeFile, mkdir } from "fs/promises";
+import { CourseFiles } from "../course";
+import { endpoints } from "./endpoints";
+import { createLogger } from "../../logger";
+
+const logger = createLogger("hmc.fetcher.utils");
+
+/**
+ * since each endpoint requires different parameter, we use this to generate all of them
+ */
+export function computeParams(term: APIv4.TermIdentifier): Params {
+    return Params.parse({
+        session: APIv4.stringifyTermIdentifier(term),
+        year: term.year.toString(),
+        catalog: `UG${
+            (term.term === APIv4.Term.fall ? term.year : term.year - 1) % 100
+        }`,
+    } satisfies Params);
+}
+
+export function computeFilePath(
+    endpoint: Endpoint,
+    term: APIv4.TermIdentifier,
+) {
+    return resolve(
```

Comment:
i think it would be a little bit better if this were `path.join` instead of `path.resolve`?

Reply:
Those two functions are practically identical since we aren't dealing with anything with a leading slash